src/utils/wallet.js

```diff
@@ -438,53 +647,266 @@ class Wallet {
+    Deploys 2/3 multisig contract with keys from contract-helper and localStorage
+    @todo check account has enough near to actually do this
 
+    const tempAccount = getTempAccount()
+    // temp account was set during create account
+    if (tempAccount && tempAccount.accountId) {
+        // console.log('temp account')
+        return {
+            temp: true,
+            balance: 0,
+            accountId: tempAccount.accountId,
+        }
+    }
+    // else load account normally from api
+    ********************************/
+    async deployMultisig() {
+        const accountData = await this.loadAccount()
+        // get multisig contract to deploy
+        const contractBytes = new Uint8Array(await fetch('/multisig.wasm').then((res) => res.arrayBuffer()))
+        console.log('contractBytes', contractBytes)
+        // get account
+        const { accountId } = accountData
         const account = this.getAccount(accountId)
+        console.log(account)
+        /********************************
+        KMS
+        ********************************/
+        // 1. get localstorage keys
         const accountKeys = await account.getAccessKeys();
-        if (!accountKeys.some(it => it.public_key.endsWith(publicKey))) {
-            await account.addKey(publicKey);
+        console.log('accountKeys', accountKeys)
+        // 2. get recovery method keys
+        const recoveryMethods = await this.getRecoveryMethods()
+        console.log('recoveryMethods', recoveryMethods)
+        const recoveryKeysED = recoveryMethods.data.map((rm) => rm.publicKey)
+        // 3. get recovery keys that are NOT seed phrases && NOT null publicKey
+        const fak2lak = recoveryMethods.data.filter(({ kind, publicKey }) => kind !== 'phrase' && publicKey !== null).map((rm) => toPK(rm.publicKey))
+        // 4. push localStorage keys that are NOT recovery keys
+        fak2lak.push(...accountKeys.filter((ak) => !recoveryKeysED.includes(ak.public_key)).map((ak) => toPK(ak.public_key)))
+        // these are the keys we need to convert to limited access keys
+        console.log('fak2lak', fak2lak)
+        // 5. get the server public key for this accountId (confirmOnlyKey)
+        const getAccessKey = await this.postSignedJson('/2fa/getAccessKey', {
+            accountId
+        })
+        if (!getAccessKey || !getAccessKey.success || !getAccessKey.publicKey) {
+            console.log('error getting publicKey from contract-helper')
+            return
         }
+        const confirmOnlyKey = toPK(getAccessKey.publicKey)
+        console.log('confirmOnlyKey', confirmOnlyKey)
+
+        // const confirm = window.confirm('deploy contract?')
+        // if (!confirm) return
+        /********************************
+        Key updates, multisig deployment and initialization 
+        ********************************/
+        const newArgs = new Uint8Array(new TextEncoder().encode(JSON.stringify({ 'num_confirmations': 2 })));
```

Comment:
makes sense to do it in near-api-js https://github.com/near/near-api-js/pull/354

Reply:
So near-api-js can encode this for me? I'm not sure what is linked here?